table/tables/tables.go

```diff
@@ -200,42 +204,66 @@ func (t *TableCommon) GetPhysicalID() int64 {
 	return t.physicalTableID
 }
 
-// Cols implements table.Table Cols interface.
-func (t *TableCommon) Cols() []*table.Column {
-	if len(t.publicColumns) > 0 {
-		return t.publicColumns
-	}
-	publicColumns := make([]*table.Column, len(t.Columns))
-	maxOffset := -1
+type getColsMode int64
+
+const (
+	_ getColsMode = iota
+	visible
+	hidden
+	full
+)
+
+func (t *TableCommon) getCols(mode getColsMode) []*table.Column {
+	columns := make([]*table.Column, 0, len(t.Columns))
 	for _, col := range t.Columns {
 		if col.State != model.StatePublic {
 			continue
 		}
-		publicColumns[col.Offset] = col
-		if maxOffset < col.Offset {
```

Comment:
Why remove these codes?

Reply:
The previous code imply that public column always before non-public column, and set the maxOffset to the last public column. When we intro hidden column, we can't guarantee the visible columns are continuous.